tests/index.js

```diff
@@ -0,0 +1,136 @@
+var assert = require('chai').assert
+var hpkp = require('../lib/hpkp')
+var Hapi = require('hapi')
+var server
+
+function createServer(port, hpkpOptions) {
+  server = new Hapi.Server()
+  server.connection({
+    port: port
+  })
+
+  server.register({
+    register: require('../index.js'),
+    options: hpkpOptions
+  }, function (err) {
+    if (err) {
+      console.error('Failed to load plugin:', err)
+    }
+  });
+
+  server.route({
+    method: 'GET',
+    path: '/',
+    handler: function (request, reply) {
+      return reply('HPKP!')
+    }
+  })
+
+  server.start()
+
+  return server
+}
+
+var sha256s = [
+  'orlando=',
+  'magic='
+]
+
+var validTestCases = [
+  {
+    name: 'should process minimum HPKP header',
+    options: {
+      maxAge: 1,
+      sha256s: sha256s
+    },
+    expectedKey: 'public-key-pins',
+    expectedHeader: 'pin-sha256="orlando="; pin-sha256="magic="; max-age=1'
+  },
+  {
+    name: 'should process with includeSubdomains',
+    options: {
+      maxAge: 1,
+      sha256s: sha256s,
+      includeSubdomains: true
+    },
+    expectedKey: 'public-key-pins',
+    expectedHeader: 'pin-sha256="orlando="; pin-sha256="magic="; max-age=1; includeSubdomains'
+  },
+  {
+    name: 'should process with reportOnly',
+    options: {
+      maxAge: 1,
+      sha256s: sha256s,
+      reportOnly: true
+    },
+    expectedKey: 'public-key-pins-report-only',
+    expectedHeader: 'pin-sha256="orlando="; pin-sha256="magic="; max-age=1'
+  },
+  {
+    name: 'should process with report-uri',
+    options: {
+      maxAge: 1,
+      sha256s: sha256s,
+      reportOnly: true,
+      reportUri: 'http://test.site'
+    },
+    expectedKey: 'public-key-pins-report-only',
+    expectedHeader: 'pin-sha256="orlando="; pin-sha256="magic="; max-age=1; report-uri="http://test.site"'
+  }
+]
+
+describe('HPKP Headers', function () {
+  validTestCases.forEach(function (testCase) {
+    var server
+    var requestOptions = {
+      method: "GET",
+      url: "/"
+    }
+    before(function () {
+      server = createServer(3000, testCase.options)
+    })
+
+    after(function () {
+      return server.stop()
+    })
+    it(testCase.name, function (done) {
+      server.inject(requestOptions, function (response) {
+        assert.equal(response.headers[testCase.expectedKey], testCase.expectedHeader)
+        done()
+      })
+    })
+  })
+})
+
+var inputTestCases = [
```

Comment:
What is the meaning of `input` in this context? I think the tests that follow might read clearer if this was something like `invalidTestCases` or `failingTestCases` instead, unless that is inaccurate for some reason.